### PR TITLE
Fix issue with yoga in X64 Release builds

### LINKED ISSUE #4122: E2ETest failure: Padding doesn't work for Views on VS 2019, v142 tools, Release x64
As part of PR #4101, the measureLayout E2E test is failing on the windows-2019 VM because the View's padding is not being applied.

I've confirmed that it repros on my 19H1+ machine, when building in VS 2019 and the v142 toolset.

We are patching Yoga.cpp in `vnext\DeforkingPatches\ReactCommon\yoga\yoga\Yoga.cpp` to add an extra measure call, apparently for the benefit of react-native-win32, but it has the side effect of messing up react-native-windows.

Removing that patch fixes the issue for react-native-windows, but may cause other unintended side-effects in react-native-windows, and of course probably messes up react-native-win32.

### PR DESCRIPTION
**Pull request by acoates-ms**
Fixes #4122 

There seems to be an issue with the v142 compiler bits which causes the code generated around CompactValue.isUndefined to not always return the correct value.

Using fp:Strict within yoga.cpp works around the issue. 
An alternative would be to fork CompactValue, and add pragma's to enable strict just around that. -- Or wait for a compiler fix. We should look at disabling this change on compiler updates to see if the compiler issue has been fixed.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4628)

### CODE REVIEW

**Comment by NickGerleman** [comment]
Could you add a comment with a link to the issue?

File: vnext/ReactCommon/ReactCommon.vcxproj [hunk 1]
```diff
@@ -155,7 +155,8 @@
     <ClCompile Include="$(YogaDir)\yoga\YGNodePrint.cpp" DisableSpecificWarnings="4244;%(DisableSpecificWarnings)" />
     <ClCompile Include="$(YogaDir)\yoga\YGStyle.cpp" DisableSpecificWarnings="4244;%(DisableSpecificWarnings)" />
     <ClCompile Include="$(YogaDir)\yoga\YGValue.cpp" />
-    <ClCompile Include="$(YogaDir)\yoga\Yoga.cpp" DisableSpecificWarnings="4244;%(DisableSpecificWarnings)" />
+    <ClCompile Condition="'$(Platform)' == 'x64'" Include="$(YogaDir)\yoga\Yoga.cpp" DisableSpecificWarnings="4244;%(DisableSpecificWarnings)" FloatingPointModel="Strict" />
```
Comment (NickGerleman) [inline vnext/ReactCommon/ReactCommon.vcxproj:158]: Could you add a comment with a link to the issue here?

Comment (asklar) [reply vnext/ReactCommon/ReactCommon.vcxproj:159]: Note that I'm only able to see a difference in we build x64 AND Release (x64 Debug looks ok)
image

**Comment by asklar** [comment]
Could you file an issue to undo this workaround when we get a fix from MSVC?

**Comment by acoates-ms** [comment]
Yes, but we may as well have debug match the strict mode to try to minimize the differences between release and debug.

**Pull request merged**